src/DataStructures/Tensor/EagerMath/DotProduct.hpp

```diff
@@ -0,0 +1,65 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions euclidean dot_product and dot_product with a metric
```

Comment:
Could you have `magnitude` call the dot_product functions? A separate commit after this one is fine

Reply:
Yep! I'll do a separate pull request for that after this one is merged.